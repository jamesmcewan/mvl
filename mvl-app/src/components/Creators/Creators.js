import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';

const List = styled.ul`
  padding: 0;
`;

const ListItem = styled.li`
  list-style-type: none;
`;

const creatorMap = creator => (
  <ListItem key={`${creator.name}-${creator.role}`}>
    {creator.name} - {creator.role}
  </ListItem>
);

const Creators = ({ items }) => {
  return <List>{items && items.map(creatorMap)}</List>;
};

Creators.propTypes = {
  items: PropTypes.array
};

export default Creators;
