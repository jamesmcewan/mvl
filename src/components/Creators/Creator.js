const Creator = (creator) => (
  <li
    className="list-none text-yellow-200"
    key={`${creator.name}-${creator.role}`}
  >
    {creator.name} - {creator.role}
  </li>
)

export default Creator
