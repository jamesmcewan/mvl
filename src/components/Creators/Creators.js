import React from 'react';
import styled from 'styled-components';
import { isArray } from 'util';
import Creator from '../Creator/Creator';

const CreatorsList = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

const CreatorsHeading = styled.h3`
  margin-top: 0;
  font-size: 22px;
`;

const Creators = ({ creators }) => (
  <div>
    <CreatorsHeading>Creative team</CreatorsHeading>
    <CreatorsList>
      {isArray(creators) &&
        creators.map(creator => (
          <Creator {...creator} key={`c-${creator.name}`} />
        ))}
    </CreatorsList>
  </div>
);

Creators.defaultProps = {
  creators: [
    {
      name: '',
      role: '',
    },
  ],
};

export default Creators;
