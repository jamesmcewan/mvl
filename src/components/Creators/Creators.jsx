import Creator from './Creator'

const Creators = ({ creators }) => (
  <ul className="p-0">
    {creators?.items?.map((creator) => (
      <Creator {...creator} key={creator.name} />
    ))}
  </ul>
)

export default Creators
