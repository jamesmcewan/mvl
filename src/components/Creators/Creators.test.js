import React from 'react';
import { mount } from 'enzyme';
import Creators from './Creators';
import 'jest-styled-components';

describe('Creators', () => {
  it('should render without crashing', () => {
    const wrapper = mount(<Creators />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<Creators />);
    expect(wrapper).toMatchSnapshot();
  });
});
