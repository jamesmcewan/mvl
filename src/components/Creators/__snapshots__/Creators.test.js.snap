// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creators should render the correct JSX 1`] = `
.c1 {
  list-style-type: none;
  margin: 0;
  padding: 0;
}

.c0 {
  margin-top: 0;
  font-size: 22px;
}

<Creators
  creators={
    Array [
      Object {
        "name": "",
        "role": "",
      },
    ]
  }
>
  <div>
    <styled.h3>
      <h3
        className="c0"
      >
        Creative team
      </h3>
    </styled.h3>
    <styled.ul>
      <ul
        className="c1"
      >
        <Creator
          key="c-"
          name=""
          role=""
        >
          <li>
             - 
          </li>
        </Creator>
      </ul>
    </styled.ul>
  </div>
</Creators>
`;
